Dockerfile.base

```diff
@@ -0,0 +1,6 @@
+# Copyright (c) 2021 Tailscale Inc & AUTHORS All rights reserved.
```

Comment:
What's this file for? Does't seem used by anything else here.

Also .base includes regular alpine and .not-base includes alpine-base. ???

Reply:
this is the extra step that i mentioned in the commit message, the idea is that we can construct and publish the base images independently.
so we build `ghcr.io/tailscale/alpine-base` once and then future tailscale container builds can rely on that without having to rebuild from `alpine`.

I'm open to alternate suggestions.

Reply:
Ah, gotcha. We're probably going to want a story soon for automating these rebuilds, otherwise we're eventually going to ship CVEs that alpine's already patched. But LGTM otherwise.